src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java

```diff
@@ -42,13 +42,13 @@ public void loadAllergiesPage() {
 
 	@Then("the system loads Allergies page")
 	public void systemLoadsAllergiesPage() {
-		assertEquals(getElement(patientHeaderId).getText(),getElement(patientHeaderId).getText());
-		assertTrue(textExists("Allergies"));
+		assertPage(allergyPage.waitForPage());
 	}
 
 	@And("a user clicks No Known Allergy button")
 	public void addNoKnownAllergy() {
 		allergyPage.addNoKnownAllergy();
+		allergyPage.waitForPage();
```

Comment:
is this line neccesary here

Reply:
refer to my comment [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/256#discussion_r763906928)

Reply:
@kdaud which action fails when you do not wait? I thought you would wait before the action that fails. Not after the previous one.

Reply:
adding known allergy seems to take longer to perform the action so as the next operation to be effective.